module-role-sync/src/main/kotlin/org/quiltmc/community/cozy/modules/rolesync/RoleSyncExtension.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package org.quiltmc.community.cozy.modules.rolesync
+
+import com.kotlindiscord.kord.extensions.annotations.DoNotChain
+import com.kotlindiscord.kord.extensions.extensions.Extension
+import com.kotlindiscord.kord.extensions.extensions.ephemeralSlashCommand
+import com.kotlindiscord.kord.extensions.extensions.event
+import com.kotlindiscord.kord.extensions.types.respond
+import com.kotlindiscord.kord.extensions.utils.hasRole
+import dev.kord.core.event.guild.MemberUpdateEvent
+import kotlinx.coroutines.flow.filter
+import org.quiltmc.community.cozy.modules.rolesync.config.RoleSyncConfig
+
+/**
+ * Module providing role sync functionality.
+ *
+ * When one of the source roles is added or removed from a user, the user is added or removed from the target role.
+ */
+public class RoleSyncExtension(
+    private val config: RoleSyncConfig
+) : Extension() {
+    override val name: String = RoleSyncPlugin.id
+
+    @OptIn(DoNotChain::class)
+    override suspend fun setup() {
+        event<MemberUpdateEvent> {
+            action {
+                // Make sure the old member is available
+                if (event.old == null) return@action
+
+                // Check if the roles have changed
+                if (event.old!!.roleIds == event.member.roleIds) return@action
```

Comment:
By convention, we always use braces with our conditionals. We certainly don't want to end up causing the next `GOTO fail;`